### PR TITLE
Implement marker heatmap

### PR DESCRIPTION
**Pull request by cosa65**
# Background
#### Link to issue 
https://biomage.atlassian.net/browse/BIOMAGE-1171

#### Link to staging deployment URL 
https://ui-martinfosco-ui396-pi177-w.scp-staging.biomage.net/

#### Links to any Pull Requests related to this
https://github.com/biomage-ltd/ui/pull/396
https://github.com/biomage-ltd/worker/pull/152

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/api/route-services/pipeline-response.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ const plotsTableService = new PlotsTablesService();
 const experimentService = new ExperimentService();
 
 const getPipelineStatus = require('../general-services/pipeline-status');
+
```
Comment (ivababukova) [inline src/api/route-services/pipeline-response.js:16]: no need for this line

File: src/api/event-services/work-request.js [hunk 2]
```diff
@@ -53,6 +53,7 @@ const handleWorkRequest = async (workRequest, socket) => {
       }
 
       const workSubmitService = new WorkSubmitService(workRequest);
+
```
Comment (ivababukova) [inline src/api/event-services/work-request.js]: no need for this line

File: src/specs/models/work-request-bodies/WorkRequestMarkerGenes.v1.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  nGenes:
+    type: number
+    description: Amount of marker genes to get
+  config:
+    type: object
+    properties:
+      resolution:
+        type: number
+        minimum: 0
+        maximum: 2
```
Comment (ivababukova) [inline src/specs/models/work-request-bodies/WorkRequestMarkerGenes.v1.yaml:18]: this sounds wrong

**Review by ivababukova — APPROVED** [review]

File: src/specs/models/work-request-bodies/WorkRequestMarkerGenes.v1.yaml (hunk 3 above)
Comment (ogibson) [reply src/specs/models/work-request-bodies/WorkRequestMarkerGenes.v1.yaml:18]: Looks like sensible values. 2 might be a bit excessive but it won't break anything. Besides, it depends on the slider for the resolution values for clustering, so it makes sense to have the same values.

**Pull request merged**